### ISSUE TITLE
[no-useless-spread] Handle more cases for `toSorted` (and the likes)

### ISSUE THREAD

**Issue by boris-petrov** [sindresorhus/eslint-plugin-unicorn#2210]
`no-useless-spread` already handles cases like `[...foo.toSorted()]` which is awesome! However, it would be nice to also handle `[...foo].toSorted()` and `[...foo].sort()`, both of which can be replaced with `foo.toSorted()`.

cc @fisker, @sindresorhus 

P.S. Similar work was done in [this PR](https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2030).

**Comment by fisker** [comment]
```js
[...foo].toSorted()
```

is not safe to fix, since `foo` can be other types like `string`, `Set`..

**Comment by fisker** [comment]
```js
[...foo].sort()
```

Should definitely go into a new rule like `prefer-array-to-sorted`.

**Comment by abrahamguo** [comment]
Wouldn't this suffer from the same problem — `foo` could still be a `string`, `Set`, etc. — and so `[...foo].sort()` cannot be reported either?

**Comment by fisker** [comment]
I think it's reasonable to make a rule to forbid `.sort()` except expression.

**Comment by sindresorhus** [comment]
This is now handled by the `no-array-sort` rule (and `no-array-reverse` for the `reverse`/`toReversed` variants), added in #2713.

As discussed here, the spread cases live in a dedicated rule rather than in `no-useless-spread`:

- `[...foo].sort()` is reported with two suggestions, since `foo` may not be an array:
  - “The spreading object is an array” → `foo.toSorted()`
  - “The spreading object is NOT an array” → `[...foo].toSorted()`
- The `allowExpressionStatement` option covers forbidding `.sort()` except as an expression statement.

The original `[...foo].toSorted()` → `foo.toSorted()` case is intentionally left alone, since (as noted above) it's only safe when `foo` is already an array, which can't be determined statically (`foo` could be a `string`, `Set`, etc.).

Closin as resolved
